### PR TITLE
add profiles for waybar and some hypr utilities

### PR DESCRIPTION
**Pull request by odomingao**

### CODE REVIEW

**Comment by roddhjav** [comment]
Thanks,

I have made a few comments. Also, in all profiles, the attachment **must** be `@{exec_path}`, otherwise it would not be recognised as it by the build system.

File: apparmor.d/groups/hypr/hyprlock [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+profile hyprlock /{,usr/}{,s}bin/hyprlock {
+  include <abstractions/base>
+  include <abstractions/dri>
+  include <abstractions/fonts>
+  include <abstractions/mesa>
+  include <abstractions/nameservice-strict>
+  include <abstractions/opencl-mesa>
+  include <abstractions/vulkan>
+  include <abstractions/vulkan-strict>
```
Comment (roddhjav) [inline apparmor.d/groups/hypr/hyprlock:19]: This is simplified as:
```suggestion
  include <abstractions/fonts>
  include <abstractions/nameservice-strict>
  include <abstractions/graphics>
```

File: apparmor.d/groups/hypr/hyprlock [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  @{bin}/unix_chkpwd mrix,
+
+  /etc/login.defs r,
+  /etc/pam.d/* r,
+  /etc/security/faillock.conf r,
+  /etc/shells r,
+  owner /etc/shadow r,
+
+  owner @{XDG_PICTURES_DIR}/** r,
+  owner @{XDG_WALLPAPERS_DIR}/** r,
```
Comment (roddhjav) [inline apparmor.d/groups/hypr/hyprlock:35]: `@{XDG_*_DIR}` are relative path, so you need `@{HOME}` or @{user_*_dirs}` vars. 

```suggestion
  owner @{HOME}/@{XDG_WALLPAPERS_DIR}/** r,
  owner @{user_pictures_dirs}/** r,
```

File: apparmor.d/groups/hypr/hyprlock [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  network netlink raw,
+
+  @{exec_path} mr,
+  @{bin}/unix_chkpwd mrix,
+
+  /etc/login.defs r,
+  /etc/pam.d/* r,
+  /etc/security/faillock.conf r,
+  /etc/shells r,
+  owner /etc/shadow r,
```
Comment (roddhjav) [inline apparmor.d/groups/hypr/hyprlock:32]: Remove this and add the `authentication` abstraction.

File: apparmor.d/groups/hypr/hyprlock [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+  owner @{XDG_PICTURES_DIR}/** r,
+  owner @{XDG_WALLPAPERS_DIR}/** r,
+
+  owner @{user_config_dirs}/hypr/hyprlock.conf r,
+
+  owner @{run}/faillock/@{user} rwk,
```
Comment (roddhjav) [inline apparmor.d/groups/hypr/hyprlock:39]: In the `authentication` abstraction.

File: apparmor.d/groups/hypr/hyprpaper [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  /usr/share/icons/** r,
+
+  owner @{XDG_WALLPAPERS_DIR}/** r,
+
+  owner @{user_config_dirs}/hypr/hyprpaper.conf r,
+
+  owner @{run}/user/*/ r,
+  owner @{run}/user/*/.hyprpaper* rw,
+  owner @{run}/user/*/hypr/*/.hyprpaper.sock w,
+  owner @{run}/user/*/hyprpaper.lock rw,
```
Comment (roddhjav) [inline apparmor.d/groups/hypr/hyprpaper:26]: This needs to be ` @{run}/user/@{uid}/`

File: apparmor.d/groups/hypr/hyprpm [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  /usr/include/** r,
+  /usr/share/git-core/** r,
+  /usr/share/pkgconfig/** r,
+
+  owner @{user_share_dirs}/hyprpm/ r,
+  owner @{user_share_dirs}/hyprpm/** rw,
+  owner @{user_share_dirs}/hyprpm/*/*/** rw,
+
+  /tmp/hyprpm/** rw,
+  /tmp/hyprpm/*/*/** rw,
```
Comment (roddhjav) [inline apparmor.d/groups/hypr/hyprpm:37]: The second rule is already included by the first.

File: apparmor.d/groups/hypr/hyprpm [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+  @{lib}/gcc/*/*/** rix,
+  @{lib}/git-core/** rix,
+
+  /usr/include/** r,
+  /usr/share/git-core/** r,
+  /usr/share/pkgconfig/** r,
+
+  owner @{user_share_dirs}/hyprpm/ r,
+  owner @{user_share_dirs}/hyprpm/** rw,
+  owner @{user_share_dirs}/hyprpm/*/*/** rw,
```
Comment (roddhjav) [inline apparmor.d/groups/hypr/hyprpm:34]:
```suggestion
  owner @{user_share_dirs}/hyprpm/{,**} rw,
```

File: apparmor.d/groups/hypr/hyprpm [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+  network inet dgram,
+  network inet stream,
+  network inet6 dgram,
+
+  @{exec_path} mr,
+
+  @{bin}/** rix,
+  @{lib}/gcc/** rix,
+  @{lib}/gcc/*/*/** rix,
```
Comment (roddhjav) [inline apparmor.d/groups/hypr/hyprpm:25]: The second rule is already included by the first.

File: apparmor.d/profiles-s-z/waybar [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+abi <abi/3.0>,
+
+include <tunables/global>
+
+@{exec_path} = @{bin}/waybar
+
+profile waybar /{,usr/}{,s}bin/waybar flags=(attach_disconnected) {
+  include <abstractions/audio>
+  include <abstractions/base>
```
Comment (roddhjav) [inline apparmor.d/profiles-s-z/waybar:13]: base must come before audio. (nitpick but, let's respect the general guideline)

File: apparmor.d/profiles-s-z/waybar [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+  @{exec_path} mr,
+
+  @{bin}/** rPUx,
+  @{user_bin_dirs}/** rPUx,
+
+  /usr/share/icons/** r,
+  /usr/share/pixmaps/** r,
+
+  owner @{HOME}/.icons/** r,
+  owner @{HOME}/.themes/** r,
```
Comment (roddhjav) [inline apparmor.d/profiles-s-z/waybar:31]: These are included in `abstractions/desktop`. Same for `owner @{user_share_dirs}/icons/** r,`

File: apparmor.d/profiles-s-z/waybar [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+
+  owner @{user_config_dirs}/waybar/config.jsonc r,
+  owner @{user_config_dirs}/waybar/style.css r,
+  owner @{user_config_dirs}/waybar/themes/** r,
```
Comment (roddhjav) [inline apparmor.d/profiles-s-z/waybar:35]: This is the config dir for itself, so you can safely simplify this by:
```
owner @{user_config_dirs}/waybar/{,**} r,
```

File: apparmor.d/groups/hypr/hyprlock (hunk 4 above)
Comment (odomingao) [reply apparmor.d/groups/hypr/hyprlock:39]: neither the faillock files nor /etc/shells seem to be present in the authentication abstraction

File: apparmor.d/profiles-s-z/waybar (hunk 9 above)
Comment (odomingao) [reply apparmor.d/profiles-s-z/waybar:13]: Got it, but the documentation only says that the order of the rules should be alphabetical, unless missed it.

File: apparmor.d/groups/hypr/hyprlock (hunk 4 above)
Comment (roddhjav) [reply apparmor.d/groups/hypr/hyprlock:39]: Good point, keep both of the rules.

File: apparmor.d/profiles-s-z/waybar (hunk 9 above)
Comment (roddhjav) [reply apparmor.d/profiles-s-z/waybar:13]: Yes, but the base abstraction is special.

**Comment by odomingao** [comment]
Thanks. I updated the profiles accordingly.

**Comment by roddhjav** [comment]
Thanks, merged.

**Pull request merged**